packages/shared/lib/core/wallet/actions/getTotalWalletBalance.ts

```diff
@@ -1,13 +1,23 @@
 import { OutputData } from '@iota/sdk/out/types'
 import { IBalance, getBicBalance } from '..'
 import { getBalance } from './getBalance'
+import { getManaBalance, getTotalAvailableMana } from '../../network'
+import { getWalletById } from '../../profile'
 
 export async function getTotalWalletBalance(walletId: string, accountOutputs: OutputData[]): Promise<IBalance> {
     const balances = await getBalance(walletId)
     const blockIssuanceCredits = await getBicBalance(walletId, accountOutputs)
     const totalWalletBic = Object.values(blockIssuanceCredits).reduce((acc, bic) => acc + Number(bic), 0)
+
+    const _selectedWallet = getWalletById(walletId)
+    let realAvailableMana = getManaBalance(balances?.mana?.available)
+    if (_selectedWallet) {
```

Comment:
This doesn't make sense, you still want to get the real available mana when the wallet is created in the onboarding. Imo, `getTotalWalletBalance` should be receiving the wallet instance, not the wallet id. Wallet ID were a mistake to be used as function param.

Reply:
There is absolutely no case where if you call getTotalWalletBalance, `_selectedWallet` should not exist, and if it didn't then the problem is elsewhere. Which in this case is that hasn't been loaded when you are in the onboarding, hence why we should be using wallet instances and not wallet ids.